web/src/main/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilter.java

```diff
@@ -260,7 +260,11 @@ private String getName(ServletContext context) {
         String name = null;
         try {
             String contextPath = context.getContextPath();
-            if (CommonUtils.isNullOrEmpty(contextPath)) {
+            TelemetryConfiguration telemetryConfiguration = TelemetryConfiguration.getActive();
+            if (telemetryConfiguration.getApplicationName() != null) {
+                name = telemetryConfiguration.getApplicationName();
+            }
+            else if (CommonUtils.isNullOrEmpty(contextPath)) {
                 URL[] jarPaths = ((URLClassLoader) (this.getClass().getClassLoader())).getURLs();
```

Comment:
This is an issue with JDK 9. The default class loader of JDK 9 is not an instance of URLClassLoader but is instead called "App-Classloader". This call to getURLs() would fail here. Do we have a work around for this ? Have you tried this on a Spring Boot Project with JDK 9?

Reply:
I haven't tried it on JDK 9, but probably `ClassLoader#getSystemResources` will do the job.